hyperjumptech/monika
# Chore/875 reduce complexity

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Closing #875

## How did you implement / how did you fix it  
1. Reduce complexity to doProbe(), remove eslint wrning
2. Add generic response processing()

## How to test  
1. use the following `config.yaml`:
```yaml
probes:
  - id: check-tcp
    name: localhost all works
    alerts:
      - query: response.status == 500
        message: tcp not accessible
    incidentThreshold: 2
    recoveryThreshold: 2

    socket:
      host: localhost
      port: 3333
      data: hello

  - id: checking-redis
    name: localhost all works
    alerts:
      - query: response.status != 200
        message: redis not available
    incidentThreshold: 2
    recoveryThreshold: 2
    redis:
      - host: 172.17.0.1
        port: 6379

  - id: http-probe-01
    name: 'test status 400'
    interval: 5
    requests:
      - url: http://httpbin.org/status/400
        method: GET
        timeout: 5000
        saveBody: false
        headers:
          Content-Type: application/json; charset=utf-8
    alerts:
      - query: response.status != 400
        message: Http Response status code is not 200!
      - query: response.time > 10000
        message: Yo too slow!
    incidentThreshold: 2
    recoveryThreshold: 2

notifications:
  - id: desktop-01
    type: desktop
```
2. Run `bin/run -c config.yaml`
3. Remove tcp/redis availability to see alerts/notifications
4. Probes should run with proper logs and alerts

# Screenshot
Monika running logs with some events
![image](https://user-images.githubusercontent.com/964106/189061681-b2b423f0-71a6-4723-bef4-c75ace9d3e6c.png)
